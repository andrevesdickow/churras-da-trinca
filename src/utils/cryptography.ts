import crypto from 'crypto';
import { isEmpty } from 'lodash';

function getBytes(str: string) {
  const bytes = [];
  for (let i = 0; i < str.length; ++i) {
    bytes.push(str.charCodeAt(i));
  }
  return bytes;
}

/**
 * Criptografa um texto
 * @param textToEncrypt texto
 * @param secretKey chave secreta
 * @returns texto criptografado
 */
export function encrypt(textToEncrypt: string, secretKey: string) {
  if (isEmpty(textToEncrypt) || isEmpty(secretKey)) {
    return '';
  }

  const key = Buffer.from(getBytes(secretKey));
  const iv = Buffer.from([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);

  const cipher = crypto.createCipheriv('aes-128-cbc', key, iv);
  let encrypted = cipher.update(textToEncrypt);
  encrypted = Buffer.concat([encrypted, cipher.final()]);
  return encrypted.toString('base64');
}
