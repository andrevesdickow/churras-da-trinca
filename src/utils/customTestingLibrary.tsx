import '../app/global.css';
import React from 'react';
import { ToastContainer } from '@/components/Toast';
import { NextThemeProvider } from '@/providers/NextThemeProvider';
import { Provider as TooltipProvider } from '@radix-ui/react-tooltip';
import { render } from '@testing-library/react';

const customRender = (ui: React.ReactElement, { ...renderOptions } = {}) => render(
  <NextThemeProvider>
    <TooltipProvider>
      {ui}
      <ToastContainer />
    </TooltipProvider>
  </NextThemeProvider>,
  renderOptions
);

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
