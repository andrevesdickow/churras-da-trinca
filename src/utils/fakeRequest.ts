export function fakeRequest<T = any>(delay: number, value: T) {
  return new Promise((resolve) => setTimeout(resolve, delay, value));
}
