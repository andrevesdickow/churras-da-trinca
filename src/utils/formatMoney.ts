/**
 * Formata valor monetÃ¡rio para real brasileiro
 * @param value valor
 * @returns Valor formatado -> R$ 1.000,00
 */
export function formatMoney(value: number) {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL'
  }).format(value);
}
