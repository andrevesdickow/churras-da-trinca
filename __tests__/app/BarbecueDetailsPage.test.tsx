import BarbecueDetailsPage from '@/app/churras/detalhes/[id]/page';
import { fireEvent, render, screen, waitForElementToBeRemoved } from '@/utils/customTestingLibrary';

jest.mock('@tanstack/react-query', () => ({
  ...jest.requireActual('@tanstack/react-query'),
  useQuery: jest.fn().mockReturnValue(({
    data: {
      id: '1',
      date: '2023-12-25',
      dateFormatted: '25/12',
      description: 'Festa de Natal',
      priceWithDrink: 100,
      priceWithoutDrink: 50
    },
    isFetching: false
  }))
}));

describe('<BarbecueDetailsPage />', () => {
  it('should render correctly Barbecue Details Page', () => {
    const { container } = render(<BarbecueDetailsPage params={{ id: '1' }} />);

    // expect(screen.getByText('25/12')).toBeInTheDocument();
    // expect(screen.getByText('Festa de Natal')).toBeInTheDocument();
    expect(screen.getByText(/Não há participantes para este churras/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should click create participant correctly in Barbecue Details Page', async () => {
    render(<BarbecueDetailsPage params={{ id: '1' }} />);

    fireEvent.submit(screen.getByRole('button'));

    expect(await screen.findByRole('alert')).toBeInTheDocument();
    expect(await screen.findByText('Nome é obrigatário')).toBeInTheDocument();

    fireEvent.input(screen.getByRole('textbox', { name: 'Nome' }), { target: { value: 'John Doe' } });

    await waitForElementToBeRemoved(() => screen.queryByRole('alert'));

    // fireEvent.submit(screen.getByRole('button'));
  });

  it('should render back button correctly in Barbecue Details Page', () => {
    render(<BarbecueDetailsPage params={{ id: '1' }} />);

    expect(screen.getByRole('link', { name: 'Voltar' })).toHaveAttribute('href', '/churras');
  });
});
