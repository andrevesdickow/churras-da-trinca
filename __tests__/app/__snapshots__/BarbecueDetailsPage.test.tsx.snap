// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarbecueDetailsPage /> should render correctly Barbecue Details Page 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="flex flex-col gap-3 p-8 rounded-3xl z-[1px] min-w-[400px] w-full h-max bg-slate-50 dark:bg-slate-800 shadow-xl"
  >
    <div
      class="flex justify-between"
    >
      <h5
        class="text-lg font-semibold"
      >
        25/12
      </h5>
      <div
        class="flex flex-row align-center gap-2"
      >
        <svg
          class="text-amber-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="9"
            cy="7"
            r="4"
          />
          <path
            d="M22 21v-2a4 4 0 0 0-3-3.87"
          />
          <path
            d="M16 3.13a4 4 0 0 1 0 7.75"
          />
        </svg>
        <span>
          0
        </span>
      </div>
    </div>
    <div
      class="flex justify-between"
    >
      <h2
        class="text-2xl font-bold"
      >
        Festa de Natal
      </h2>
      <div
        class="flex flex-row align-center gap-2"
      >
        <svg
          class="text-amber-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M16 8h-6a2 2 0 1 0 0 4h4a2 2 0 1 1 0 4H8"
          />
          <path
            d="M12 18V6"
          />
        </svg>
        <span>
          R$Â 0,00
        </span>
      </div>
    </div>
    <h3
      class="font-bold text-amber-900 dark:text-slate-50 text-center mt-6"
    >
      Participantes
    </h3>
    <p
      class="text-slate-600 dark:text-slate-300 text-center"
    >
      NÃ£o hÃ¡ participantes para este churras ðŸ«¤
    </p>
    <form
      class="flex flex-col justify-between mt-4"
    >
      <div
        class="flex items-center gap-4"
      >
        <div
          class="relative z-0 mb-4 flex-1"
        >
          <input
            class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
            id="name"
            name="name"
            placeholder=" "
            role="textbox"
            type="text"
            value=""
          />
          <label
            class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
            for="name"
          >
            Nome
          </label>
        </div>
        <div
          class="flex flex-col mb-2"
        >
          <div
            class="flex gap-2 items-center"
          >
            <button
              aria-checked="false"
              class="peer h-4 w-4 shrink-0 rounded-sm border border-amber-300 ring-offset-white focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-amber-300 data-[state=checked]:text-white dark:border-amber-300 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 dark:data-[state=checked]:bg-amber-300 dark:data-[state=checked]:text-white"
              data-state="unchecked"
              id="withDrink"
              role="checkbox"
              type="button"
              value="false"
            />
            <input
              aria-hidden="true"
              name="withDrink"
              style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
              tabindex="-1"
              type="checkbox"
              value="false"
            />
            <label
              class="text-sm text-gray-500 dark:text-gray-300 hover:cursor-pointer"
              for="withDrink"
            >
              Com bebida?
            </label>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 rounded-3xl text-sm font-medium uppercase ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 bg-slate-950 text-white hover:bg-slate-950/90 dark:bg-slate-950 dark:text-white dark:hover:bg-slate-950/50 h-10 px-4 py-2"
        type="submit"
      >
        Adicionar participante
      </button>
    </form>
    <a
      class="text-amber-400 hover:text-amber-500 transition-colors text-center text-sm"
      href="/"
    >
      Voltar
    </a>
  </div>
</div>
`;
